### PR TITLE
adds a failing test to check that regions and global.dist should be different

### PR DESCRIPTION
**Pull request by bwlang**
regions and global dist should be different
re #105

### CODE REVIEW

**Comment by brentp** [comment]
you can fix with this:
```
diff --git a/mosdepth.nim b/mosdepth.nim
index 015a13b..5ff0bc1 100644
--- a/mosdepth.nim
+++ b/mosdepth.nim
@@ -592,7 +592,7 @@ proc main(bam: hts.Bam, chrom: region_t, mapq: int, eflag: uint16, iflag: uint16
   if not open(fh_summary, prefix & ".mosdepth.summary.txt", fmWrite):
     stderr.write_line("[mosdepth] could not open file:", prefix & ".mosdepth.summary.txt")
 
-  if region != "" and not open(fh_region_dist, prefix & ".mosdepth.region.dist.txt", fmWrite):
+  if region != "" and not region.isdigit and not open(fh_region_dist, prefix & ".mosdepth.region.dist.txt", fmWrite):
     stderr.write_line("[mosdepth] could not open file:", prefix & ".mosdepth.dist.txt")
 
   if region != "":
```

**Comment by bwlang** [comment]
that leaves fh_regoin_dist null.
I could add more checks to avoid trying to write to it later and avoid writing the extra file at all.

But I think region_dist should contain the average coverage in the region size.  Does that make sense?
Not sure if this is the direction you want to go... it's a bit weird that 1000 = average, but providing a bed file wtih 1000 bp intervals would not.  Maybe it's better to add a command line flag to report the average in the region (like goleft indexcov, but with flexible window size)

**Comment by brentp** [comment]
no. I don't think this change makes sense. now I am confused about what you want. if you give regions, it *will* give you the mean of each region in the regions.bed file. I think it's fine to leave this or use the diff I added.

**Comment by bwlang** [comment]
this is what things look like with the change:
image

**Comment by brentp** [comment]
and what mosdepth command-line are you using for one of those?

**Comment by bwlang** [comment]
both are from the same command:
./mosdepth t_regions -n -b 1000 tests/empty-tids.bam
per region is t_regions.mosdepth.region.dist.txt
per base is t_regions.mosdepth.global.dist.txt

**Comment by bwlang** [comment]
I'm trying to get a more flexible version of goleft indexcov

**Comment by brentp** [comment]
ok, now I get what you're trying to do. but I'm not sure that's what mosdepth should do. I would instead use post-processing on regions.bed.gz to create your own mean .dist.txt

**Comment by bwlang** [comment]
what would you think about adding a command line flag 
-r --average-region report average coverage by region (instead of per-base)

**Comment by brentp** [comment]
you might check out: https://github.com/brwnj/covviz
I think it will take a regions.bed.gz and draw an indexcov like plot.

**Comment by brentp** [comment]
no thanks on that option. this seems easily solved by post-processing the regions.bed.gz and I don't like to add more options

**Comment by bwlang** [comment]
covviz looks cool - but I work at a company.

**Comment by brentp** [comment]
I'll have another look at this tomorrow. I suppose if the change only applies to --by $number, then it's not surprising.

**Comment by brentp** [comment]
I'm OK with this once the tests are passing.

File: mosdepth.nim [hunk 1]
```diff
@@ -755,7 +755,9 @@ Common Options:
 
   -t --threads <threads>     number of BAM decompression threads [default: 0]
   -c --chrom <chrom>         chromosome to restrict depth calculation.
-  -b --by <bed|window>       optional BED file or (integer) window-sizes.
+  -b --by <bed|window>       optional BED file or (integer) window-sizes. 
+                             integer window-sizes will produce region.dist.txt files summarizing
```
Comment (brentp) [inline mosdepth.nim]: don't change the command-line help. but update readme and CHANGES.md

**Comment by bwlang** [comment]
Maybe this is more in line with what you're thinking.

**Comment by brentp** [comment]
cheers!

**Pull request merged**